compute_tools/src/config.rs

```diff
@@ -157,3 +159,82 @@ where
 
     res
 }
+
+fn get_rsyslog_pid() -> Option<String> {
+    let output = Command::new("pgrep")
+        .arg("rsyslogd")
+        .output()
+        .expect("Failed to execute pgrep");
+
+    if !output.stdout.is_empty() {
+        let pid = std::str::from_utf8(&output.stdout)
+            .expect("Invalid UTF-8 in process output")
+            .trim()
+            .to_string();
+        Some(pid)
+    } else {
+        None
+    }
+}
+
+// Start rsyslogd with the specified configuration file
+// If rsyslogd is running and restart it if necessary
+fn restart_rsyslog(rsyslog_conf_path: &str) -> Result<()> {
+    let pid = get_rsyslog_pid();
+    if let Some(pid) = pid {
+        info!("rsyslogd is already running with pid: {}", pid);
+        // kill the running rsyslogd
+        let _ = Command::new("kill").arg("-9").arg(&pid).output();
+    }
+
+    let _ = Command::new("/usr/sbin/rsyslogd")
+        .arg("-f")
+        .arg(rsyslog_conf_path)
+        .output()
+        .context("Failed to start rsyslogd")?;
+    info!("Started rsyslogd.");
+
+    // DEBUG ONLY double check if rsyslogd is running
+    let pid = get_rsyslog_pid();
+    if pid.is_none() {
+        return Err(anyhow::anyhow!("Failed to start rsyslogd"));
+    }
+    info!("rsyslogd started successfully with pid: {}", pid.unwrap());
+
+    Ok(())
+}
+
+pub fn configure_rsyslog(log_directory: &str, tag: &str, remote_endpoint: &str) -> Result<()> {
```

Comment:
We need to support constructing rsyslog config in two scenarios: 

1. For HIPAA projects we configure rsyslog during compute startup
2. When user enables logs export we need to reconfigure rsyslog on the fly

With your changes (1) is supported. As for (2) is compute spec stored somewhere so that we could reference it when reloading the config?

How about having a more structured configuration?

```suggestion
struct RsyslogConfig {
    HipaaHost: Option<String>,
    LogsExportHost: Option<String>,
}

pub fn configure_rsyslog(log_directory: &str, conf: &RsyslogConfig) -> Result<()> {
```

Reply:
This PR is only HIPAA related.
I want to focus on it now for simplicity and add support for 2nd stream of logs later.

It will also require some extra setup of postgres logs / pgaudit.